packages/@lwc/template-compiler/package.json

```diff
@@ -21,7 +21,8 @@
     },
     "devDependencies": {
         "@types/estree": "0.0.48",
-        "@types/he": "^1.1.1"
+        "@types/he": "^1.1.1",
+        "@types/source-map": "0.5.2"
```

Comment:
`source-map` is a transitive dependency and the version used is pretty old. The newer versions have typescript support but unfortunately it does not satisfy the version required. I could override and specify the latest version of source-map in our package.json

Reply:
Seems safer to me to import the types for the old package than try to override with the newer package.

Reply:
I don't understand why using `source-map` is needed here? Where do we use `source-map` in the template compiler?

Reply:
`yarn why source-map` shows multiple packages rely on it

<details><summary>click here</summary>

```
=> Found "source-map@0.6.1"
info Has been hoisted to "source-map"
info Reasons this module exists
   - "workspace-aggregator-b5b0bfa9-908a-412a-b018-95c6433d59ee" depends on it
   - Hoisted from "_project_#integration-karma#karma#source-map"
   - Hoisted from "_project_#jest#@jest#core#@jest#reporters#source-map"
   - Hoisted from "_project_#rollup-plugin-terser#terser#source-map-support#source-map"
   - Hoisted from "_project_#jest#@jest#core#@jest#transform#source-map"
   - Hoisted from "_project_#integration-karma#karma-coverage#istanbul-lib-source-maps#source-map"
   - Hoisted from "_project_#jest#@jest#core#jest-runtime#@jest#source-map#source-map"
   - Hoisted from "_project_#lerna#@lerna#version#@lerna#conventional-commits#conventional-changelog-core#conventional-changelog-writer#handlebars#source-map"
   - Hoisted from "_project_#jest#@jest#core#jest-config#jest-environment-jsdom#jsdom#escodegen#source-map"
```
</details>

I guess it's technically not the template compiler that needs it? So maybe we should just put it in the top-level package.json?

Reply:
`@types/source-map` was not required after all. Running `yarn upgrade` seems to have fixed the source-map version at 0.6.1 which has typescript support.